service/debugger/locations.go

```diff
@@ -303,8 +304,33 @@ func (loc *NormalLocationSpec) FileMatch(path string) bool {
 	return partialPathMatch(loc.Base, path)
 }
 
-func tryMatchRelativePathByProc(expr, debugname, file string) bool {
-	return len(expr) > 0 && expr[0] == '.' && file == path.Join(path.Dir(debugname), expr)
+func (loc *NormalLocationSpec) tryMatchFullPath(d *Debugger, file string) bool {
```

Comment:
It seems like the only thing this method uses from `*Debugger` is `d.processArgs`. Let's just pass that in by itself instead of passing in the entire `Debugger` struct.